gensim/models/word2vec.py

```diff
@@ -921,6 +951,39 @@ def worker_loop():
     def clear_sims(self):
         self.syn0norm = None
 
+    def update_weights(self):
+        """
+        Copy all the existing weights, and reset the weights for the newly
+        added vocabulary.
+        """
+        logger.info("updating layer weights")
+        newsyn0 = empty((len(self.vocab), self.vector_size), dtype=REAL)
+
+        # copy the weights that are already learned
+        for i in xrange(0, len(self.syn0)):
+            newsyn0[i] = deepcopy(self.syn0[i])
+
+        # randomize the remaining words
+        for i in xrange(len(self.vocab), len(newsyn0)):
+            # construct deterministic seed from word AND seed argument
+            self.syn0[i] = self.seeded_vector(self.index2word[i] + str(self.seed))
```

Comment:
Seems this should start at the *old* vocab length, and the newly-seeded vectors should be landing in the `newsyn0`? (Doubt this could cause segfaults as out-of-bounds access should be caught here, but true cause might be something similar elsewhere...)